### PR TITLE
Bind missing Hedera operations in EVM version 0.34

### LINKED ISSUE #4748: Create2OperationSuite fails with the new EVM version
[This test](https://github.com/hashgraph/hedera-services/blob/899df96b3b63fe16e33afe6538e299bf1386b921/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/Create2OperationSuite.java#L439) tries to execute a `CREATE2` operation a couple of times with the same arg list: [wronglyDeployTwice](https://github.com/hashgraph/hedera-services/blob/899df96b3b63fe16e33afe6538e299bf1386b921/test-clients/src/main/resource/contract/contracts/Create2Factory/Create2Factory.sol#L64).

The test fails, but not as one may expect - because a contract already exists at the target address - rather, it fails because we're trying to use the full `callValue()` amount for both contract deployments:
1. the first CREATE2 instruction succeeds, and the contract balance is decreased to 0
2. the second CREATE2 instruction causes the balance check in [AbstractRecordingCreateOperation](https://github.com/hashgraph/hedera-services/blob/899df96b3b63fe16e33afe6538e299bf1386b921/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/operation/AbstractRecordingCreateOperation.java#L113) to fail 

The EVM then executes an `ExtCodeSizeOperation`. There is a difference in how we handle this between EVM versions `0.30` and `0.34` because `ContractsV_0_34Module` no longer binds `HederaExtCodeSizeOperation` and instead the Besu `ExtCodeSizeOperation` is invoked, which doesn't execute the [addressCheck](https://github.com/hashgraph/hedera-services/blob/899df96b3b63fe16e33afe6538e299bf1386b921/hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/contracts/operations/HederaExtCodeSizeOperation.java#L65).
- EVM 0.30 tries to execute an address check for address (0x0) and fails with `INVALID_SOLIDITY_ADDRESS`
- EVM 0.34 does not execute the address check and so the execution proceeds and the final status is `CONTRACT_REVERT_EXECUTED`

### PR DESCRIPTION
**Pull request by stoqnkpL**
**Description**:

- Binds the missing operations:`HederaBalanceOperation`, `HederaExtCodeCopyOperation`, `HederaExtCodeHashOperation`, and `HederaExtCodeSizeOperation` in `ContractsV_0_34Module`.
- Updates the Create2Factory contract.
- Updates the EVM version to 0.34 in testing property files.

**Related issue(s)**:

Fixes #4748 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/ContractsV_0_34Module.java [hunk 1]
```diff
@@ -186,6 +186,46 @@ static Operation bindStaticCallOperation(
                 gasCalculator, sigsVerifier, addressValidator, precompiledContractMap);
     }
 
+    @Provides
+    @Singleton
+    @IntoSet
+    @ContractsModule.V_0_30
```
Comment (stoqnkpL) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/ContractsV_0_34Module.java:192]: Should this be V_0_34

Comment (dimitar-dinev) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/ContractsV_0_34Module.java]: It should, fixed now

**Review by dimitar-dinev — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Pull request merged**